lib/library.rb

```diff
@@ -0,0 +1,52 @@
+class Library
+  attr_reader :name,
+              :books,
+              :authors,
+              :checked_out_books,
+              :popularity
+  def initialize(name)
+    @name = name
+    @books = []
+    @authors = []
+    @checked_out_books = []
+    @popularity = Hash.new { |popularity, book| popularity[book] = 0 }
+  end
+
+  def add_author(author)
+    @authors << author
+    author.books.each do |book|
```

Comment:
What would happen here if the author wrote another book after they were added to the library? We wouldn't have a way to add that book independently, this is why it's a good idea to calculate the books based off of the @authors array.